### PR TITLE
Blake2s app access

### PR DESCRIPTION
**Pull request by mchack-work**
- New MMIO to store the address of firmware's `blake2s()` function.
- Changes to firmware to fill it in when starting.
- Tests in testfw.

Still needs doc update.

### CODE REVIEW

**Review by quite — APPROVED** [review]
I've also run this towards the qemu branch, and on hw.

File: hw/application_fpga/fw/tk1/main.c [hunk 1]
```diff
@@ -174,6 +175,9 @@ int main()
 			// CDI = hash(uds, hash(app), uss)
 			compute_cdi(digest, use_uss, uss);
 
+			// Let the app know the function adddress for blake2s()
+			*fw_blake2s_addr = (uint32_t) blake2s;
```
[the remark this answers is not in the document]
Comment (mchack-work) [reply hw/application_fpga/fw/tk1/main.c]: Let's init it at once at start.

**Comment by quite** [comment]
> Probably want to squash

Run fmt on all the C code as well. And moving that end-of-line comment for the blake2s_addr setup to its own line would make it more readable perhaps.

**Comment by mchack-work** [comment]
Squashed and merged.

**Pull request closed without merging**